### DISCUSSION TITLE
Installation Help on Windows

### DISCUSSION THREAD

**Discussion by tomleetomlee** [Clay-foundation/model#189] [Q&A]
Hello,

Seeking help with installing on a local device.

I have been following the instructions provided at https://clay-foundation.github.io/model/installation.html under the Getting Started Installation section. Seems like the instructions pertain to Linux or Mac OS. Is that the case? 

Nevertheless I try the advanced install instructions and I receive an error about conda-lock.yml not working with Win-64. Any input or tips would be much appreciated on this.

Finally, is there a necessary PyTorch version for installation of the model on a Windows OS? Seems like the environment.yml and conda-lock files only include PyTorch specific versions for Linux & Mac OS. 

Thank you!

**Comment by weiji14** [comment]
Hi @tomleetomlee, thanks for raising this issue. Are you interested in running the Clay model in training mode, or inference (prediction) mode. Also, do you have Windows Subsystem for Linux (WSL) set up on your computer, or are you interested in running things natively on Windows?

> I have been following the instructions provided at https://clay-foundation.github.io/model/installation.html under the Getting Started Installation section. Seems like the instructions pertain to Linux or Mac OS. Is that the case?
> 
> Nevertheless I try the advanced install instructions and I receive an error about conda-lock.yml not working with Win-64. Any input or tips would be much appreciated on this.
>
> Finally, is there a necessary PyTorch version for installation of the model on a Windows OS? Seems like the environment.yml and conda-lock files only include PyTorch specific versions for Linux & Mac OS.

Yes, the `conda-lock.yml` only works with Linux and macOS for now, because [Pytorch from the conda-forge channel](https://anaconda.org/conda-forge/pytorch/files?version=2.1.2) doesn't have Windows builds yet (see outstanding issue at https://github.com/conda-forge/pytorch-cpu-feedstock/issues/32). You could install [Pytorch from the pytorch channel](https://anaconda.org/pytorch/pytorch/files?version=2.1.2), but it is not something we've tested out yet. We can work on adding support for running model inference on the CPU perhaps on Windows, but training the model with a GPU on a Windows system might be too hard, in which case I'd recommend WSL.

**Comment by tomleetomlee** [comment]
Thank you @weiji14 for your response. 

_**I would like to run both in training mode and inference (prediction) mode.**_

I have WSL on my computer. I have no preference to local or through WSL. _**Can you provide guidance on both?**_

Got it on the PyTorch. Thank you.

**Reply by weiji14** [reply]
WSL2 would probably be better if you're intending to run training. I actually don't have experience with WSL, but maybe @MaceGrim can provide advice on this? I think he's running WSL on VSCode somehow?

**Reply by MaceGrim** [reply]
Thanks for the mention.

When I tried to do this ~1.5 months ago, I wasn't able to get things running on Windows, BUT, WSL made things super smooth. Just go through the given Linux instructions through your WSL terminal and it should work.

When you're ready to develop things, the ```code .``` command in WSL will download and launch VSCode's GUI. It'll run within WSL, so you'll have access to the ```claymodel``` environment and other tools you might've installed within WSL.

Let me know if any of that is unclear or if you get stuck somewhere.

**Reply by tomleetomlee** [reply]
Thank you very much both. I will try that out and let you know if I run into any issues. Thank you.

**Reply by MaceGrim** [reply]
No problem! Forgot to mention, my WSL is running Ubuntu 22.04.3 LTS.
